### ISSUE TITLE
python3 download button is missing

### ISSUE THREAD

**Issue by bitdust** [python/pythondotorg#865]
![python3fail](https://cloud.githubusercontent.com/assets/6072743/11690930/d08cd898-9ed3-11e5-9494-9e33fb9b3d12.jpg)

windows8.1 x64 chrome-46

**Comment by willingc** [comment]
@bitdust Thanks for the report. :+1:

**Comment by malemburg** [comment]
Hmm, it works for me. Could you please clear your cache and recheck ? This was an issue yesterday, but I had fixed it (or at least think I did).

**Comment by berkerpeksag** [comment]
It didn't worked for me earlier, but it works now.

**Comment by malemburg** [comment]
On 09.12.2015 23:45, Berker Peksag wrote:

> It didn't worked for me earlier, but it works now.

I have a feeling that the settings on the release records
are either not well documented or not well understood
(probably both :-)).

AFAIK, a script is used to set up the releases. Perhaps something
is wrong with that script. I just don't know where to look for
it.

## 

Marc-Andre Lemburg
http://www.malemburg.com/

**Comment by ned-deily** [comment]
It's the release manager's responsibility to get the fields set. For Python 3.5, you should be asking Larry Hastings.

**Comment by berkerpeksag** [comment]
> AFAIK, a script is used to set up the releases. Perhaps something
> is wrong with that script. I just don't know where to look for
> it.

I think the script is add-to-pydotorg.py: https://hg.python.org/release/file/tip/add-to-pydotorg.py

I think we can add some checks to the downloads app to prevent reports like this in the future so the release manager can easily fix the problem in the first place.

**Comment by malemburg** [comment]
On 10.12.2015 13:34, Berker Peksag wrote:

> > AFAIK, a script is used to set up the releases. Perhaps something
> > is wrong with that script. I just don't know where to look for
> > it.
> 
> I think the script is add-to-pydotorg.py: https://hg.python.org/release/file/tip/add-to-pydotorg.py
> 
> I think we can add some checks to the downloads app to prevent reports like this in the future so the release manager can easily fix the problem in the first place.

Thanks. It seems that this part needs an update:

```
    download_button = 'tar.xz' in rfile or
                      'macosx10.6.dmg' in rfile or
                      'macosx10.6.pkg' in rfile or
                      ('.msi' in rfile and not 'amd64' in rfile),
```

since it does cover the .exe installers used for Windows
in 3.5 (there are no MSI installers anymore).

## 

Marc-Andre Lemburg
http://www.malemburg.com/

**Comment by bitdust** [comment]
good job! i just find it works for me

**Comment by willingc** [comment]
Great. Thanks @bitdust for letting us know that it is now working for you.

**Comment by larryhastings** [comment]
Some helpful soul already fixed this for 3.5.1; I've fixed the automation Marc referred to.

**Comment by malemburg** [comment]
@larryhastings Yes, we fixed the settings manually in the database. Thanks for fixing the script. It now reads:

```
    download_button = 'tar.xz' in rfile or
                      'macosx10.6.dmg' in rfile or
                      'macosx10.6.pkg' in rfile or
                      (rfile.endswith(('.msi', '.exe'))
                        and not ('amd64' in rfile or 'webinstall' in rfile)),
```

If we ever have both MSI and .exe installers, this will may need fixing again to make the choice deterministic :-)
